src/messages/embed.ts

```diff
@@ -0,0 +1,257 @@
+	/**
+	 * The embed url.
+	 */
+	public url?: string;
+
+	/**
+	 * The embed bar color.
+	 */
+	public color?: number;
+
+	/**
+	 * The timestamp of the embed.
+	 */
+	public timestamp?: string;
+
+	/**
+	 * The embed thumbnail data.
+	 */
+	public thumbnail?: APIEmbedThumbnail;
+
+	/**
+	 * The embed image data.
+	 * @since 0.0.1
+	 */
+	public image?: APIEmbedImage;
+
+	/**
+	 * Received video data.
+	 * @since 0.0.1
+	 */
+	public video?: APIEmbedVideo;
+
+	/**
+	 * The embed author data.
+	 * @since 0.0.1
+	 */
+	public author?: APIEmbedAuthor;
+
+	/**
+	 * Received data about the embed provider.
+	 * @since 0.0.1
+	 */
+	public provider?: APIEmbedProvider;
+
+	public footer?: APIEmbedFooter;
+
+	public constructor(data: APIEmbed = {}) {
+		this.title = data.title;
+		this.description = data.description;
+		this.url = data.url;
+		this.color = data.color;
+		this.thumbnail = data.thumbnail;
+		this.image = data.image;
+		this.video = data.video;
+		this.author = data.author;
+		this.provider = data.provider;
+		this.footer = data.footer;
+
+		if (data.timestamp) this.timestamp = new Date(data.timestamp).toISOString();
+		if (data.fields) this.fields = data.fields;
+	}
+
+	/**
+	 * The date displayed on this embed
+	 */
+	public get createdAt(): Date | null {
+		return this.timestamp ? new Date(this.timestamp) : null;
+	}
+
+	/**
+	 * The accumulated length for the embed title, description, fields, footer text, and author name
+	 */
+	public get length(): number {
+		return (
+			(this.title?.length ?? 0) +
+			(this.description?.length ?? 0) +
+			(this.fields.length >= 1
+				? this.fields.reduce((prev, curr) => prev + curr.name.length + curr.value.length, 0)
+				: 0) +
+			(this.footer?.text.length ?? 0) +
+			(this.author?.name?.length ?? 0)
+		);
+	}
+
+	/**
+	 * Adds a field to the embed (max 25).
+	 * @param name The name of this field
+	 * @param value The value of this field
+	 * @param inline If this field will be displayed inline
+	 */
+	public addField(name: string, value: string, inline = false): this {
+		return this.addFields({ name, value, inline });
+	}
+
+	/**
+	 * Adds fields to the embed (max 25).
+	 * @param fields The fields to add
+	 */
+	public addFields(...fields: APIEmbedField[]): this {
+		this.fields.push(...Embed.normalizeFields(...fields));
+		return this;
+	}
+
+	/**
+	 * Removes, replaces, and inserts fields in the embed (max 25).
+	 * @param index The index to start at
+	 * @param deleteCount The number of fields to remove
+	 * @param fields The replacing field objects
+	 */
+	public spliceFields(index: number, deleteCount: number, ...fields: APIEmbedField[]): this {
+		this.fields.splice(index, deleteCount, ...Embed.normalizeFields(...fields));
+		return this;
+	}
+
+	/**
+	 * Sets the author of this embed.
+	 * @param name The name of the author
+	 * @param iconURL The icon URL of the author
+	 * @param url The URL of the author
+	 */
+	public setAuthor(name: string, iconURL?: string, url?: string): this {
```

Comment:
Maybe `iconURL` and `url` should be in an options object so that we can use `.setAuthor('name', { url: 'some URL' })` instead of `.setAuthor('name', undefined, 'some URL')`?

Reply:
Done in https://github.com/discordjs/builders/pull/11/commits/877f1c655d11e25771989210d24a37163f3b3736 ♥